### PR TITLE
Facets search query

### PR DESCRIPTION
**Pull request by LHBruneton-C2C**
Add / remove facet to query on item selected change
Facets are a part of filters.
Selected items and block are filtered from their path.

### CODE REVIEW

File: libs/search/src/lib/state/reducer.spec.ts [hunk 1]
```diff
@@ -104,4 +106,93 @@ describe('Search Reducer', () => {
...
+            aggregations: { someKey: 'someValue' },
+          },
+        },
+        action
+      )
+      expect(state.results.aggregations).toEqual(ES_FIXTURE_AGGS_RESPONSE_TERM)
+    })
+  })
+
+  describe('SetResultsAggregations action', () => {
```
Comment (fgravin) [inline libs/search/src/lib/state/reducer.spec.ts:128]:
```suggestion
  describe('SetConfigAggregations action', () => {
```

**Review by fgravin — CHANGES_REQUESTED** [review]
As discussed, the object `params.facets` in the search state is redundant.
Facet actions should update `params.filter`, I think I missed that from the previous PR.
Also, your facet object was missing the index key (eg `tag`).

File: libs/search/src/lib/facets/facets-container/facets-container.component.ts [hunk 2]
```diff
@@ -30,10 +32,85 @@ export class FacetsContainerComponent implements OnInit {
...
+            path,
+            selected
+          )
+          this.store.dispatch(new SetFilters(newFilters))
+        })
+      )
+      .subscribe()
+  }
+
+  findSelectedPaths(filters) {
```
Comment (fgravin) [inline libs/search/src/lib/facets/facets-container/facets-container.component.ts:64]: those functions diserve tests

**Review by fgravin — APPROVED** [review]
Thanks for the work.
I'm going to test it and merge it, maybe I will do some adjustments but it's fine overall.

**Comment by fgravin** [comment]
replaced by https://github.com/geonetwork/geonetwork-ui/pull/79

File: libs/search/src/lib/elasticsearch/elasticsearch.service.ts [hunk 3]
```diff
@@ -19,16 +18,124 @@ export class ElasticsearchService {
...
+            },
+          ],
         },
       },
     }
     return payload
   }
+
+  fromFiltersToQuery(filters) {
+    const clone = JSON.parse(JSON.stringify(filters))
```
[the remark this answers is not in the document]
Comment (LHBruneton-C2C) [reply libs/search/src/lib/elasticsearch/elasticsearch.service.ts:43]: I needed to do a deep copy to remove immutability from the state object. Does this syntax remove it too ?

[the remark this answers is not in the document]
Comment (fgravin) [reply libs/search/src/lib/elasticsearch/elasticsearch.service.ts:43]: No it's not recursive. But in this particular case you cloned it just to remove the `any` property, seems to work with my changes.

**Pull request closed without merging**